src/ui/GenioWindow.cpp

```diff
@@ -1705,40 +1705,40 @@ GenioWindow::_FileOpen(BMessage* msg)
 
 		// new file to load..
 		selectTabInfo.AddBool("caret_position", true);
-		int32 index = fTabManager->CountTabs();
+		int32 index = fTabManager->SelectedTabIndex() + 1;
 		Editor* editor = _AddEditorTab(&ref, index, &selectTabInfo);
 
+		LogTrace("New index: %d, selected index: %d", index, fTabManager->SelectedTabIndex());
+
 		// TODO: using assert() is not nice, try to handle
 		// this gracefully if possible
 		assert(index >= 0);
 		assert(editor != nullptr);
 
 		status = editor->LoadFromFile();
 		if (status != B_OK) {
+			LogError("Failed loading file: %s", ::strerror(status));
 			continue;
 		}
 
 		editor->ApplySettings();
 
-		// Let's assign the right project to the Editor
-		for (int32 index = 0; index < GetProjectBrowser()->CountProjects(); index++) {
-			ProjectFolder * project = GetProjectBrowser()->ProjectAt(index);
-			_TryAssociateOrphanedEditorsWithProject(project);
-		}
-
-		fTabManager->SelectTab(index, &selectTabInfo);
+		// Select the newly added tab
+		int32 newIndex = fTabManager->TabForView(editor);
+		fTabManager->SelectTab(newIndex, &selectTabInfo);
 
 		BMessage noticeMessage(MSG_NOTIFY_EDITOR_FILE_OPENED);
 		noticeMessage.AddString("file_name", editor->FilePath());
 		SendNotices(MSG_NOTIFY_EDITOR_FILE_OPENED, &noticeMessage);
 
-		LogInfo("File open: %s [%d]", editor->Name().String(), fTabManager->CountTabs() - 1);
+		LogInfo("File open: %s [%d]", editor->Name().String(), newIndex - 1);
 	}
 
-	// If at least 1 item or more were added select the first
-	// of them.
-	if (nextIndex < fTabManager->CountTabs())
```

Comment:
Aren't we going to support this anymore? Imagine you drop 5 files in the editor, in the current version you select the first file on the left, in this change the last one on the right. Same when you open the application and you automatically reaload the open files.. 
Not sure if I care much to keep it to be honest.. for the dropped one I don't care for the ones loaded at startup maybe the right solution should be to also reload what was the selected one and reselect it.. but it's a different feature..